### PR TITLE
Add changes to use Keycloak with dojot 

### PR DESCRIPTION
**Pull request by mprevide**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
dojot/dojot#1986

* **Other information**:
certgen:  dojot/dojot#2170

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,118 @@
...
+
+KEYCLOAK_REALM_SSL_MODE=EXTERNAL
+```
+
+Then change the file `letsencrypt-nginx/nginx-challenge.conf`, replacing __`<YOUR DOMAIN>`__ tag by your registered domain.
+
+After changing `letsencrypt-nginx/nginx-challenge.conf`, run the command below to run the service that will be used by Let's Encrypt to verify your registered domain.
+
+``` sh
+sudo docker-compose up -d letsencrypt-nginx
```
Comment (danielapetito) [inline README.md:109]: is not it necessary inform docker-compose-challenge.yml ?

Comment (mprevide) [reply README.md:109]: Not anymore

**Review by danielapetito — COMMENTED** [review]
in the docker-compose.yml, is not "development" the correct image ?

**Comment by mprevide** [comment]
> in the docker-compose.yml, is not "development" the correct image ?

I'm still going to change it, so I left this PR as a draft. As other related PRs have not yet been merged. I left the names of future images commented, but I left the temporary name of the images so that it is possible to upload the compose.

**Comment by mprevide** [comment]
In the part of creation self signed certificate, the code used to gereate the certs is in https://github.com/dojot/dojot/tree/feature/certgen-self-rsa/tools/certgen

**Comment by danielapetito** [comment]
in the  docker-compose.yml, is not the correct image development ?

**Comment by mprevide** [comment]
> in the docker-compose.yml, is not the correct image development ?

Este PR não vai ser margeado na development por enquanto,  e para que seja possivel executa-lo é necessário usar estas imgs que coloquei. Qdo chegar o momento deste PR entrar development, vocês irão ter que trocar as imgs que faltam para a tag ``${DOJOT_VERSION}``

File: README.md [hunk 2]
```diff
@@ -85,13 +148,139 @@ sudo crontab -e
...
+On Debian-based Linux distributions, you can install these prerequisites by running:
+
+``` sh
+sudo apt install git openssl
+```
+
+After installing the prerequisites if necessary, download the tool to generate certificates [certgen](https://github.com/dojot/dojot/tree/development/tools/certgen) (to see other possible parameters go to [documents](https://github.com/dojot/dojot/tree/development/tools/certgen#readme)):
+
+```sh
+git clone --single-branch --branch development https://github.com/dojot/dojot.git cert-tools-temp && cp -r cert-tools-temp/tools/certgen certgen && rm -rf cert-tools-temp
```
Comment (mprevide) [inline README.md:169]: To test use https://github.com/dojot/dojot/tree/feature/certgen-self-rsa/tools/certgen

Comment (mprevide) [reply README.md:169]: to test use feature/certgen-self-rsa instead of development  for now

**Review by mprevide — COMMENTED** [review]
use certgen from https://github.com/dojot/dojot/pull/2170

File: keycloak/keycloak.config.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+
+set -eu -o pipefail
+
+KEYCLOAK_HOST=${KEYCLOAK_HOST:-"http://keycloak:8080/auth"}
+KEYCLOAK_CREATE_REALM=${KEYCLOAK_CREATE_REALM:-"admin"}
+KEYCLOAK_MASTER_USER=${KEYCLOAK_MASTER_USER:-"admin"}
+KEYCLOAK_MASTER_PASSWORD=${KEYCLOAK_MASTER_PASSWORD:-"admin"}
```
Comment (cfrancisco) [inline keycloak/keycloak.config.sh:8]: hm... it sounds that will be loaded with admin/admin user. This being the case, could show a warning when the user loads the GUI/login screen.

File: kong/kong.config.sh [hunk 4]
```diff
@@ -151,15 +159,15 @@ addAuthToEndpoint "history"
...
-# service: backstage
+createEndpoint "data-manager-export" "http://data-manager:3000/"  '"/export"' "false"
+addAuthToEndpoint "data-manager-export"
 
-createEndpoint "backstage_graphql_auth" "http://backstage:3005/"  '"/graphql-auth"' "false"
+# service: backstage
 
-createEndpoint "backstage_graphql" "http://backstage:3005/"  '"/graphql"' "false"
-addAuthToEndpoint "backstage_graphql"
+createEndpoint "backstage" "http://backstage:3000"  '"/backstage"' "false"
```
Comment (cfrancisco) [inline kong/kong.config.sh:170]: hm... we should check the backstage's endpoint in GUI/GUIv2.

**Review by cfrancisco — COMMENTED** [review]
We also need to change all docker-compose images to $[DOJOT_VERSION] again, beyond the previous comments.

**Pull request closed without merging**